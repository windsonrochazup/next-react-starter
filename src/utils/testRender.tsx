import { render, RenderOptions } from '@testing-library/react'
import { ReactElement } from 'react'
import { ThemeProvider } from 'styled-components'
import Token from '../styles/theme'

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: Omit<RenderOptions, 'queries'> = {}
) => render(<ThemeProvider theme={Token}>{ui}</ThemeProvider>, renderOptions)

export * from '@testing-library/react'
export { customRender as render }
