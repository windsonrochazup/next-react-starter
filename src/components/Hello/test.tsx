import { render, screen } from 'utils/testRender'

import { Hello } from './index'

describe('<Hello />', () => {
  it('should render the heading', () => {
    const { container } = render(<Hello />)

    expect(screen.getByRole('heading', { name: /Hello/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the heading with blue color', () => {
    render(<Hello />)

    expect(screen.getByRole('heading', { name: /Hello/i })).toHaveStyleRule('color', 'blue')
  })
})
