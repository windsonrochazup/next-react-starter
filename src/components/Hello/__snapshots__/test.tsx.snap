// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hello /> should render the heading 1`] = `
.c0 {
  color: blue;
}

<h1
  class="c0"
>
  Hello World ðŸš€ 
</h1>
`;
